File: typescript-supplychain/vulnerable15.ts
import { exec } from "child_process";
import OpenAI from "openai";
import express, { Request, Response } from "express";

const app = express();
app.use(express.json());
const client = new OpenAI();

app.post("/install-package", async (req: Request, res: Response) => {
  const task: string = req.body.task;
  const completion = await client.chat.completions.create({
    model: "gpt-4",
    messages: [{ role: "user", content: `What npm package should I use for: ${task}? Reply with only the package name.` }],
  });
  const packageName: string = completion.choices[0].message.content!.trim();
  exec(`npm install ${packageName}`, (err) => {
    if (err) return res.status(500).json({ error: err.message });
    res.json({ installed: packageName });
  });
});

File: python-supplychain/vulnerable15.py
import subprocess
import openai
from flask import Flask, request, jsonify

app = Flask(__name__)
client = openai.OpenAI()

@app.post("/setup-deps")
def setup_deps():
    model_task = request.json["task"]
    response = client.chat.completions.create(
        model="gpt-4",
        messages=[{"role": "user", "content": f"List pip packages for: {model_task}. One per line, no versions."}],
    )
    packages = response.choices[0].message.content.strip().splitlines()
    for pkg in packages:
        subprocess.run(["pip", "install", pkg.strip()], check=True)
    return jsonify({"installed": packages})
